ddl/ddl_db_test.go

```diff
@@ -185,6 +185,9 @@ func (s *testDBSuite) TestMySQLErrorCode(c *C) {
 	s.testErrorCode(c, sql, tmysql.ErrNoSuchTable)
 	sql = "alter table test_error_code_succ add column `a ` int ;"
 	s.testErrorCode(c, sql, tmysql.ErrWrongColumnName)
+	s.tk.MustExec("create table test_on_update (c1 int, c2 int, c3 int );")
+	sql = "alter table test_on_update add column c4 int on update current_timestamp;"
+	s.testErrorCode(c, sql, tmysql.ErrInvalidOnUpdate)
```

Comment:
Could we add a test like `create table t(c int on update current_timestamp);` It can cover line331 in ddl_api.go